app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -45,22 +55,50 @@ public static String generateRandomId() {
     return sb.toString();
   }
 
-  public static void addDocument(String key, Map document, String collection)
-      throws RuntimeException {
-    DatabaseWrapper.getInstance()
-        .firestore
-        .collection(collection)
-        .document(key)
-        .set(document)
-        .addOnSuccessListener(aVoid -> {})
-        .addOnFailureListener(
-            e -> {
-              throw new RuntimeException(e);
-            });
+  static <T extends Document> void addDocument(T document, String collection) {
+    getCollectionReference(collection).document(document.getId()).set(document);
+  }
+
+  static <T extends Document> void removeDocument(String key, String collection) {
+    getCollectionReference(collection).document(key).delete();
+  }
+
+  static <T extends Document> void updateDocument(
+      String key, Map<String, Object> updates, String collection) {
+    getCollectionReference(collection).document(key).update(updates);
+  }
+
+  static <T extends Document> CompletableFuture<T> getDocument(
+      String key, Class<T> cls, String collection) throws RuntimeException {
+    Task<DocumentSnapshot> getTask = getCollectionReference(collection).document(key).get();
+    CompletableFuture<DocumentSnapshot> getFuture = new TaskToFutureAdapter<>(getTask);
+
+    return getFuture.thenApply(
+        documentSnapshot -> {
+          if (documentSnapshot.exists()) {
+            return documentSnapshot.toObject(cls);
+          } else {
+            throw new RuntimeException(String.format("Document %s does not exist ", key));
```

Comment:
Is throwing a runtime exception the best way to handle this case?

Reply:
Do you have any ideas as to what I can do instead?

Reply:
I'll take care of this in the next PR.

Reply:
I mean, it's ok to do that but you should probably catch the exception in the callers (CollectionWrapper or FavorUtil). Otherwise, you can simply log the error: this case shouldn't happen if the methods are used in the correct way so logging might be enough.